govcd/metadata_test.go

```diff
@@ -381,3 +381,476 @@ func (vcd *TestVCD) Test_AddMetadataOnMediaRecord(check *C) {
 	check.Assert(metadata.MetadataEntry[0].Key, Equals, "key")
 	check.Assert(metadata.MetadataEntry[0].TypedValue.Value, Equals, "value")
 }
+
+func (vcd *TestVCD) Test_MetadataOnAdminCatalogCRUD(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	var catalogName string = check.TestName()
+
+	org, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	check.Assert(org, NotNil)
+
+	catalog, err := vcd.org.CreateCatalog(catalogName, catalogName)
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+	AddToCleanupList(catalogName, "catalog", org.AdminOrg.Name, catalogName)
+
+	adminCatalog, err := org.GetAdminCatalogByName(catalogName, true)
+	check.Assert(err, IsNil)
+	check.Assert(adminCatalog, NotNil)
+
+	metadata, err := adminCatalog.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(metadata, NotNil)
+
+	testData := map[string]struct {
+		value string
+		kind  string
+	}{
+		"key1": {
+			value: "123",
+			kind:  types.MetadataNumberValue,
+		},
+		"key2": {
+			value: "value2",
+			kind:  types.MetadataStringValue,
+		},
+	}
+
+	for k, v := range testData {
+		err := adminCatalog.AddMetadataEntry(v.kind, k, v.value)
+		check.Assert(err, IsNil)
+
+	}
+
+	// Check if metadata was added correctly
+	metadata, err = adminCatalog.GetMetadata()
+	check.Assert(err, IsNil)
+	check.Assert(metadata, NotNil)
+	check.Assert(len(metadata.MetadataEntry), Equals, len(testData))
+
+	for k, v := range testData {
+		var foundEntry *types.MetadataEntry
+		for _, entry := range metadata.MetadataEntry {
+			if entry.Key == k {
+				foundEntry = entry
+				break
+			}
+		}
+		check.Assert(foundEntry, NotNil)
+		// check.Assert(foundEntry.TypedValue.XsiType, Equals, v.kind) // known issue, check filter_helpers.go guessMetadataType() func
+		check.Assert(foundEntry.TypedValue.Value, Equals, v.value)
+	}
+
+	for k := range testData {
+		err = adminCatalog.DeleteMetadataEntry(k)
+		check.Assert(err, IsNil)
+	}
+
+	err = adminCatalog.Delete(true, true)
+	check.Assert(err, IsNil)
+}
+
+func (vcd *TestVCD) Test_ReadMetadataOnCatalog(check *C) {
```

Comment:
For that specific case, the name is ok, since from adminCatalog you can do CRUD operations, whilst from just catalog you can just read, hence the test name.